### ISSUE TITLE
Rounded image

### ISSUE THREAD

**Issue by vddgil** [janko/image_processing#72]
Hello,

Here what's I'm trying to achieve: I would like to have rounded avatars directly on file, css is not an option for me because
those avatars will be rendered in email signatures.

I'm using rails with active storage.

Any help would be awesome !

Gil

**Comment by janko** [comment]
I'm not much of an ImageMagick guru, I'm afraid. If you're able to find the ImageMagick command that does what you want, I can help you translate it into ImageProcessing API.

**Comment by vddgil** [comment]
Hey @janko. Found this on the internets ...
`convert -size 150x150 xc:none -fill image.jpg -draw "circle 75,75 75,1" circle_thumb.png.`

**Comment by vddgil** [comment]
Key @janko, here is a command that seems to wokr, can't figure out how to build that with image processing :(

`convert input.png \
        -gravity Center \
        \( -size 200x200 \
           xc:Black \
           -fill White \
           -draw 'circle 100 100 100 1' \
           -alpha Copy \
        \) -compose CopyOpacity -composite \
        -trim output.png`

**Comment by janko** [comment]
This should do it:

```rb
ImageProcessing::MiniMagick
  .source(file)
  .gravity("Center")
  .stack { |magick|
    magick
      .size("200x200")
      .xc("Black")
      .fill("White")
      .draw("circle 100 100 100 1")
      .alpha("Copy")
  }
  .compose("CopyOpacity")
  .composite
  .trim
  .call
```

**Comment by vddgil** [comment]
Ok, I was missing the stack command, but it doesn't seems to work well with active storage :(
I tried to pass those options without success ...
`{ gravity: "Center", stack: { size: "180x180", xc: "Black", fill: "White", draw: "circle 90 90 90 1", alpha: "Copy" }, compose: "CopyOpacity", composite: true, trim: true }`

**Comment by janko** [comment]
Yes, that's the limitation of ActiveStorage's hash-based approach and not having the ability to define macros – you cannot represent everything with a hash 🤷🏻‍♂️

**Comment by janko** [comment]
I've just released MiniMagick 4.11.0 that allows passing arguments to `#stack`, which can then be used with Active Storage:

```rb
ImageProcessing::MiniMagick
  .source(file)
  .gravity("Center")
  .stack(
    size: "200x200",
    xc: "Black",
    fill: "White".
    draw: "circle 100 100 100 1",
    alpha: "Copy",
   )
  .compose("CopyOpacity")
  .composite
  .trim
  .call
```

**Comment by zorar4ik** [comment]
@janko Hi! I just bumped here just to ask what about `Vips` analogue of command you've written above? So we would have both variants for rounded images and maybe later it would be possible to move it directly to Wiki. Thanks

P.S atm I use this example https://github.com/libvips/libvips/issues/1763#issuecomment-669205198
